### DISCUSSION TITLE
Packages depending on eachother

### DISCUSSION THREAD

**Discussion by sepehr500** [vercel/turborepo#6415] [Help]
Hi guys. I am trying to figure out how to have two packages import from each other. For background, I have two microservices in my monorepo that make calls to eachother, and I want them to be able to import types from eachother for type safety. The obvious problem is circular dependencies. The the packages reference eachother in the package.json, I get a circular dependency. Is there a better way to do this?

**Comment by anthonyshew** [comment]
Circular dependency graphs are most often solved by a rethink of the way your packages by some minor re-architecting of your packages. Do you have a minimal reproduction you could share?

**Reply by sepehr500** [reply]
Thanks for the response.  I have two microservices that make REST requests to eachother. For type safety when making REST calls, I want them to be be able to import types from eachother from where the API calls are defined.  (sorry I don't have a minimal reproduction). I am trying to avoid making a "shared types" subrepo given that the API types should be collocated with the API definition itself. Maybe people who use GRPC have a similar issue? I'm not as familiar with GRPC tho.
